### PR TITLE
Add lexer for YANG 1.1

### PR DESCRIPTION
**Pull request by gribok**
Based on issue #1407, I created a new lexer for YANG. 

### Features
* Found and added additional keywords based on [RFC7950](https://tools.ietf.org/html/rfc7950#page-184) to lexer
* Seperating keywords in groups, based on BNF definition of [RFC7950](https://tools.ietf.org/html/rfc7950#page-184) for maintenance
* Highlighting of constants, like `true`, `false` or `deprecated` with special token
* Highlighting of namespace-prefix, like `type namespace-prefix:counter64;`. `namespace-prefix` will have special token. 
For example `namespace urn:ieee:std:802.1AX:yang:ieee802-dot1ax;` will have no special token
* Prefix-highlighting should be the complex regex pattern of all pattern. Due to that, I created test cases to validate prefix-highlighting
* Create example yang file for testing
* No breaked rules in pylint

### CODE REVIEW

**Comment by gribok** [comment]
Simplify ruleset for YANG lexer based on PR rouge-ruby/rouge#1458, which I have created.
If you accept my PR, please squash the commits.

**Comment by Anteru** [comment]
Hi, sorry for not getting to this earlier. This is looking good, one question on the example file -- where is that from? Could there be a license incompatibilities with it?

**Comment by gribok** [comment]
> Hi, sorry for not getting to this earlier. This is looking good, one question on the example file -- where is that from? Could there be a license incompatibilities with it?

Good point. I changed the example file to an own created file.

File: pygments/lexers/yang.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+from pygments.lexer import (RegexLexer, bygroups, words)
+from pygments.token import (Text, Token, Name, String, Comment,
+                            Number)
+
+__all__ = ['YangLexer']
+
+class YangLexer(RegexLexer):
+    """
+    Lexer for `YANG <https://tools.ietf.org/html/rfc7950/>`_, based on RFC7950
+    """
```
Comment (Anteru) [inline pygments/lexers/yang.py:21]: Please add a `.. versionadded:: 2.7` line here.

Comment (gribok) [reply pygments/lexers/yang.py:23]: Done. Thanks for support.

**Comment by Anteru** [comment]
Merged, thanks for your contribution!

**Pull request merged**